pkg/master/master.go

```diff
@@ -285,6 +290,14 @@ func (cfg *Config) Complete() CompletedConfig {
 		c.ExtraConfig.MasterEndpointReconcileTTL = DefaultEndpointReconcilerTTL
 	}
 
+	if c.ExtraConfig.EndpointReconcilerConfig.Nodename == nil {
+		if hostname, err := nodeutil.GetHostname(""); err == nil {
```

Comment:
this is not necessarily the node name, I don't think we can default to this

Reply:
this also seems to assume that all masters are always hosted on nodes

Reply:
I use the same mechanics as in kube-proxy https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-proxy/app/server_others.go#L114, except `hostnameOverride` param. 

I can propose to add new cmd option to the kube-apiserver `--hostname-override`, similar to kube-proxy https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-proxy/app/server.go#L162

Reply:
I still don't think we can default this on, since it assumes every API server is on a node represented by a node name. cc @kubernetes/sig-network-pr-reviews for impact of reporting a nodename in the endpoints for the kubernetes service if no such node existed.